components/scripts/lib/info.sh

```diff
@@ -237,5 +237,16 @@ exit_with_return_code() {
     exit 3
   fi
 
+  # shellcheck disable=SC2034 # not all of the scripts have the fail if not optimized CLI argument
+  if [ -n "${_arg_fail_if_not_optimized+x}" ]; then
+    # If the script has the "fail if not optimized CLI argument", then see if the build is optimized
+    local executed_avoidable_tasks
+    executed_avoidable_tasks=$(( executed_cacheable_num_tasks[1] ))
+    if [[ "${_arg_fail_if_not_optimized}" == "on" ]] && (( executed_avoidable_tasks > 0 )); then
+      print_bl
+      die "ERROR: Build is not optimized" -2
```

Comment:
I'm not familiar on the convention for failure codes (negative, positive, etc.). We can discuss it.

Reply:
I did some quick research on this, and it looks like it is more common to use positive exit codes instead of negative ones. Exit code 1 is usually reserved for "general failures" (like how we're currently using exit code -1). Otherwise, the exit codes are generally different across the board for different applications.

As such, I propose we use "1" for general failures, and "2" if the build is not fully cacheable.